lib/expander.js

```diff
@@ -1785,6 +1785,114 @@ var Controllers = {
       },
     },
   },
+
+  // https://cdn-shop.adafruit.com/datasheets/ads1115.pdf
+  ADS1115: {
+    ADDRESSES: {
+      value: [0x48, 0x49, 0x4A, 0x4B]
+    },
+    REGISTER: {
+      value: {
+        CONFIG: 0x01,
+        READ: 0x00,
+        PIN: [0xC1, 0xD1, 0xE1, 0xF1],
+        PIN_DATA: 0x83,
+      }
+    },
+    initialize: {
+      value: function(opts) {
+        var state = priv.get(this);
+
+        state.reading = false;
+
+        this.address = opts.address || this.ADDRESSES[0];
+
+        opts.address = this.address;
+        this.io.i2cConfig(opts);
+
+        Object.assign(this.MODES, this.io.MODES);
+
+        for (var i = 0; i < 4; i++) {
+          this.pins.push({
+            supportedModes: [
+              this.MODES.ANALOG
+            ],
+            mode: 1,
+            value: 0,
+            report: 0,
+            analogChannel: i
+          });
+          this.analogPins.push(i);
+        }
+
+        this.name = "ADS1115";
+        this.isReady = true;
+
+        this.emit("connect");
+        this.emit("ready");
+      }
+    },
+    normalize: {
+      value: function(pin) {
+        if (typeof pin === "string" && pin[0] === "A") {
+          return +pin.slice(1);
+        }
+        return pin;
+      }
+    },
+    pinMode: {
+      value: function(pin, mode) {
+        this.pins[pin].mode = mode;
+      }
+    },
+    analogRead: {
+      value: function(pin, callback) {
+        var state = priv.get(this);
+        this.pins[pin].report = 1;
+
+        var ready = false;
+
+        this.on("analog-read-" + pin, callback);
+
+        // Since this operation will read all 4 pins,
+        // it only needs to be initiated once.
+        if (!state.reading) {
+          state.reading = true;
+
+          // CONVERSION DELAY
+          var delay = function () {
+            setTimeout(function () {
+              ready = true;
+            }, 8);
+          };
+
+          this.io.i2cWrite(this.address, this.REGISTER.CONFIG, [this.REGISTER.PIN[pin], this.REGISTER.PIN_DATA]);
+          delay();
+
+          this.io.i2cRead(this.address, this.REGISTER.READ, 2, function(data) {
+            if (ready) {
+              ready = false;
+
+              var newPin = pin === this.pins.length - 1 ? 0 : pin + 1;
+
+              this.io.i2cWrite(this.address, this.REGISTER.CONFIG, [this.REGISTER.PIN[newPin], this.REGISTER.PIN_DATA]);
+
+              var value = (data[0] << 8) + data[1];
+              this.pins[pin].value = value;
+
+              if (this.pins[pin].report) {
+                this.emit("analog-read-" + pin, value);
+              }
+
+              pin = newPin;
+
+              delay();
+            }
+          }.bind(this));
+        }
+      }
+    },
+  }
```

Comment:
I wish this had tests, but I think I would like to write them so I can get familiar with this chip. Thanks for this _excellent work_